### PR TITLE
Added term usage CSV export task, refs #11058

### PR DESCRIPTION
**Pull request by mcantelon**
Added a task to export, as CSV, terms, in a given taxonomy, that have been
assocated with information objects.

### CODE REVIEW

File: lib/flatfile/config/QubitTerm.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Columns that map directly to term properties
+direct:
+  - id
+  - parentId
+  - taxonomyId
+  - code
+  - sourceCulture
+  - name
+  - culture
```
Comment (jraddaoui) [inline lib/flatfile/config/QubitTerm.yml]: Do we really need this file?

File: lib/task/export/csvExportTermUsageTask.class.php [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      $criteria->add(QubitTaxonomy::ID, $options['taxonomy-id']);
+
+      if (null === QubitTaxonomy::getOne($criteria))
+      {
+        throw new sfException('Invalid taxonomy-id.');
+      }
+
+      return $options['taxonomy-id'];
+    }
+    else if(isset($options['taxonomy-name']))
```
Comment (jraddaoui) [inline lib/task/export/csvExportTermUsageTask.class.php]: We normally use elseif and it's missing the space after the if.

File: lib/task/export/csvExportTermUsageTask.class.php [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    /*
+
+    Idea for future:
+
+    1. Count uses with SELECT DISTINCT t.id, COUNT(object_id) AS uses, r.object_id FROM term t INNER JOIN object_term_relation r ON r.term_id=t.id WHERE t.taxonomy_id=35 GROUP BY(id) ORDER BY t.id;
+    2. Add uses to "usage" export config and change QubitEport so setColumn will work before exportResource
+    3. LEFT JOIN could include unused terms so we could count those too
+    4. Add --mode=used|both|unused-only ("used" as default) to flip between LEFT and INNER join (and maybe add criteria for specific modes)
+
+    */
```
Comment (jraddaoui) [inline lib/task/export/csvExportTermUsageTask.class.php]: Could you remove the empty lines at the beginning and end and add one after the comment block? And I prefer the full block ...

/*
 *
 *
 */

File: lib/task/export/csvExportTermUsageTask.class.php [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    $format = 'SELECT DISTINCT t.id FROM %s t INNER JOIN %s r ON r.term_id=t.id WHERE t.taxonomy_id=? ORDER BY t.id';
```
Comment (jraddaoui) [inline lib/task/export/csvExportTermUsageTask.class.php]: I think we should add another JOIN in this query with the object table, and check the class_name column to find only relations with QubitInformationObject.

**Review by jraddaoui — CHANGES_REQUESTED** [review]
Looks great Mike! Just a few notes.

**Comment by jraddaoui** [comment]
It looks like the commit message refers to a client ticket, could you create a public one explaining the task, etc. And I'm not sure if @fiver-watson would like to add some docs about it.

**Comment by mcantelon** [comment]
Updated and ready for another review... thanks!

**Comment by mcantelon** [comment]
(Oops, didn't mean to close PR.)

**Comment by MikeFE** [comment]
LGTM :+1: -- don't forget to make the merge commit msg point to a public ticket when merging into mainline AtoM (can use Redmine's duplicate ticket functionality if one doesn't exist).

**Comment by mcantelon** [comment]
Thanks Mike!

**Pull request merged**